test/core/storage/trie/trie_db/trie_batch_test.cpp

```diff
@@ -0,0 +1,191 @@
+};
+
+class MockDb : public test::MapDb {
+ public:
+  MOCK_METHOD2(put, outcome::result<void>(const Buffer &, const Buffer &));
+
+  // to retain the ability to call the actual implementation of put from the
+  // superclass
+  outcome::result<void> true_put(const Buffer &key, const Buffer &value) {
+    return MapDb::put(key, value);
+  }
+};
+
+/**
+ * @given an empty trie
+ * @when putting some entries into it using a batch
+ * @then all inserted entries are accessible from the trie
+ */
+TEST_F(TrieBatchTest, Put) {
+  auto batch = trie->batch();
+  FillSmallTrieWithBatch(*batch);
+
+  for (auto &entry : data) {
+    EXPECT_OUTCOME_TRUE(res, trie->get(entry.first));
+    ASSERT_TRUE(res.empty());
+  }
+  EXPECT_OUTCOME_TRUE_void(r, batch->commit());
+  for (auto &entry : data) {
+    EXPECT_OUTCOME_TRUE(res, trie->get(entry.first));
+    ASSERT_EQ(res, entry.second);
+  }
+
+  EXPECT_OUTCOME_TRUE_void(__, trie->put("102030"_hex2buf, "010203"_hex2buf));
+  EXPECT_OUTCOME_TRUE_void(_, trie->put("104050"_hex2buf, "0a0b0c"_hex2buf));
+  EXPECT_OUTCOME_TRUE(v1, trie->get("102030"_hex2buf));
+  ASSERT_EQ(v1, "010203"_hex2buf);
+  EXPECT_OUTCOME_TRUE(v2, trie->get("104050"_hex2buf));
+  ASSERT_EQ(v2, "0a0b0c"_hex2buf);
+}
+
+/**
+ * @given a small trie
+ * @when removing some entries from it using a batch
+ * @then removed entries are no longer in the trie, while the rest of them stays
+ */
+TEST_F(TrieBatchTest, Remove) {
+  auto batch = trie->batch();
+  FillSmallTrieWithBatch(*batch);
+
+  EXPECT_OUTCOME_TRUE_void(_, batch->remove(data[2].first));
+  EXPECT_OUTCOME_TRUE_void(__, batch->remove(data[3].first));
+  EXPECT_OUTCOME_TRUE_void(___, batch->remove(data[4].first));
+
+  EXPECT_OUTCOME_TRUE_void(r, batch->commit());
+
+  ASSERT_FALSE(trie->contains(data[2].first));
+  ASSERT_FALSE(trie->contains(data[3].first));
+  ASSERT_FALSE(trie->contains(data[4].first));
+  ASSERT_TRUE(trie->contains(data[0].first));
+  ASSERT_TRUE(trie->contains(data[1].first));
+}
+
+/**
+ * @given a small trie
+ * @when replacing an entry in it (put a data with an existing key) using a
+ * batch
+ * @then the value on the key is updated
+ */
+TEST_F(TrieBatchTest, Replace) {
+  auto batch = trie->batch();
+  EXPECT_OUTCOME_TRUE_void(__, batch->put(data[1].first, data[3].second));
+  EXPECT_OUTCOME_TRUE_void(r, batch->commit());
+  EXPECT_OUTCOME_TRUE(res, trie->get(data[1].first));
+  ASSERT_EQ(res, data[3].second);
+}
+
+/**
+ * @given a batch with commands
+ * @when clearing it
+ * @then no batch commands will be executed during commit, as there are none
+ * left after clear()
+ */
+TEST_F(TrieBatchTest, Clear) {
+  testing::StrictMock<MockPolkadotTrieDb> mock_trie{};
+  PolkadotTrieBatch batch{mock_trie};
+
+  // this method is called when batch tries to apply its actions, which is
+  // undesired in this case
+  EXPECT_CALL(mock_trie, getRootHash()).Times(0);
+
+  EXPECT_OUTCOME_TRUE_void(r1, batch.put("123"_buf, "111"_buf));
```

Comment:
maybe use ASSERT_TRUE ?
```suggestion
  ASSERT_TRUE(batch.put("123"_buf, "111"_buf));
```

Reply:
https://github.com/soramitsu/kagome/pull/140#discussion_r291500533

Reply:
ok, then maybe create a task (technical debt) for refactoring all EXPECT_OUTCOME_... macro definitions and discuss it later.

Reply:
Refactored for EXPECT_OUTCOME_TRUE_1, left a task to create an alias which will be easier to find by name. PRE-209.